test/HostActivationTests/GivenThatICareAboutMultilevelSharedFxLookup.cs

```diff
@@ -0,0 +1,421 @@
+            AddAvailableSharedFxVersions(_cwdSharedFxBaseDir, "9999.0.1");
+
+            // Version: 9999.0.0
+            // CWD: 9999.0.1, 9999.0.0-dummy1
+            // User: 9999.0.2, 9999.0.0-dummy2
+            // Exe: 9999.0.0, 9999.0.0-dummy0
+            // Expected: 9999.0.1 from cwd
+            dotnet.Exec(appDll)
+                .WorkingDirectory(_currentWorkingDir)
+                .EnvironmentVariable("COREHOST_TRACE", "1")
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdErrContaining(Path.Combine(_cwdSelectedMessage, "9999.0.1"));
+
+            // Remove dummy folders from user dir
+            DeleteAvailableSharedFxVersions(_userSharedFxBaseDir, "9999.0.2", "9999.0.0-dummy2");
+        }
+
+        [Fact]
+        public void SharedFxLookup_Must_Not_Roll_Forward_If_Framework_Version_Is_Specified_Through_Argument()
+        {
+            var fixture = PreviouslyBuiltAndRestoredPortableTestProjectFixture.
+                Copy();
+
+            var dotnet = fixture.BuiltDotnet;
+            var appDll = fixture.TestProject.AppDll;
+
+            // Add some dummy versions
+            AddAvailableSharedFxVersions(_cwdSharedFxBaseDir, "9999.0.1", "9999.0.0-dummy0");
+            AddAvailableSharedFxVersions(_userSharedFxBaseDir, "9999.0.2", "9999.0.0-dummy2");
+            AddAvailableSharedFxVersions(_exeSharedFxBaseDir, "9999.0.0", "9999.0.3", "9999.0.0-dummy3");
+
+            // Version: 9999.0.0 (through --fx-version arg)
+            // CWD: 9999.0.1, 9999.0.0-dummy0
+            // User: 9999.0.2, 9999.0.0-dummy2
+            // Exe: 9999.0.0, 9999.0.3, 9999.0.0-dummy3
+            // Expected: 9999.0.0 from exe dir
+            dotnet.Exec("--fx-version", "9999.0.0", appDll)
+                .WorkingDirectory(_currentWorkingDir)
+                .EnvironmentVariable("COREHOST_TRACE", "1")
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdErrContaining(Path.Combine(_exeSelectedMessage, "9999.0.0"));
+
+            // Version: 9999.0.0-dummy1 (through --fx-version arg)
+            // CWD: 9999.0.1, 9999.0.0-dummy0
+            // User: 9999.0.2, 9999.0.0-dummy2
+            // Exe: 9999.0.0, 9999.0.3, 9999.0.0-dummy3
+            // Expected: no compatible version
+            dotnet.Exec("--fx-version", "9999.0.0-dummy1", appDll)
+                .WorkingDirectory(_currentWorkingDir)
+                .EnvironmentVariable("COREHOST_TRACE", "1")
+                .CaptureStdOut()
+                .CaptureStdErr()
+                .Execute()
+                .Should()
+                .Fail()
+                .And
+                .HaveStdErrContaining("It was not possible to find any compatible framework version");
```

Comment:
I'm not sure. That's not something related to my tests. Other tests (like PortableAppActivation tests) have the same behavior. If the execution fails, then it will print the failing output (I think it happens even if there is no assertion being checked).

I believe it's purposely done like that.